rfcs/scope-of-wpt.md

```diff
@@ -0,0 +1,60 @@
+# RFC 215: Defining the Scope of web-platform-tests
+
+### Summary
+
+Define the scope of web-platform-tests to specifications which might reasonably be called "web specifications"
+and which are intended to be implemented by web browsers.
+
+Additionally, define what "tentative" might reasonably be used for.
+
+This is intended to codify existing practice,
+rather than represent any change.
+
+
+### Details
+
+Historically, what's been in scope for web-platform-tests has been somewhat nebulous,
+and largely based on a common understanding of participants,
+especially those on the Core Team.
+We should put this on surer footing,
+and make it easier for others to understand what is and isn't reasonable to add.
+
+The primary aim of web-platform-tests is to test
+[web specifications](https://github.com/w3c/browser-specs/blob/11a71b738f5e41f9239fdcd2074153388c8c6b8b/README.md#spec-selection-criteria)
+[intended to be implemented by browsers](https://github.com/w3c/browser-specs/blob/11a71b738f5e41f9239fdcd2074153388c8c6b8b/README.md#categories),
+both RFC 2119 "must" and RFC 2119 "should"
+implementation requirements.
+
+A specification does not need to have cross-vendor support for its tests to be included in web-platform-tests.
+(XXX: Should we set a bar of one-vendor support or are we okay with zero-vendor support?)
+
+Additionally, other features can be added as
+[tentative](https://web-platform-tests.org/writing-tests/file-names.html#:~:text=.tentative,-%3A%20)
+tests:
+
+ * Web browser behavior currently being explored via an
+   [explainer](https://tag.w3.org/explainers/),
+   but without a specification yet written.
+   (XXX: require prototyping to have started?)
+
+ * Historic but unspecified features in web browsers,
+   especially where major browsers are interoperable,
+   where there exists a long-term intention to specify them.
+   (Note: this excludes features which have been deliberately removed from specifications.
+    These are explicitly out of scope.)
```

Comment:
This arguably doesn't actually match up with what [the documentation currently says](https://web-platform-tests.org/writing-tests/file-names.html#:~:text=.tentative,-%3A%20):

> Indicates that a test makes assertions not yet required by any specification, or in contradiction to some specification. This is useful when implementation experience is needed to inform the specification. It should be apparent in context why the test is tentative and what needs to be resolved to make it non-tentative.

The "in contradiction to some specification" is kinda troubling — this implies that if some spec change is made, with no expectation of implementations changing behaviour any time soon, you could change the existing tests to be tentative and add new non-tentative tests for the new behaviour. IMO, that should be against policy — anything that is tentative should be on a path to becoming non-tentative. We should probably be explicit in the scope of what can be tentative that everything which _is_ tentative should be on a path to becoming non-tentative.

I don't think the two things listed here are a complete list of cases; we also have cases:

 * where there are WG resolutions not yet applied to the spec making tests tentative (e.g., [`/css/css-overflow/line-clamp/line-clamp-005.tentative.html`](https://github.com/web-platform-tests/wpt/blob/73fd679bba753839932a62a8b33988e79494f088/css/css-overflow/line-clamp/line-clamp-005.tentative.html) is an example of this, with https://github.com/w3c/csswg-drafts/issues/7708#issuecomment-1758081281),
 * where there is a proposal to change the spec to some other behaviour (e.g., [`svg-aam/role/role-img.tentative.html`](https://github.com/web-platform-tests/wpt/blob/73fd679bba753839932a62a8b33988e79494f088/svg-aam/role/role-img.tentative.html#L17-L19), with https://github.com/w3c/svg-aam/issues/32),
 * where there is a proposed addition to the spec (e.g., [`/html/dom/elements/global-attributes/the-anchor-attribute-001.tentative.html`](https://github.com/web-platform-tests/wpt/blob/73fd679bba753839932a62a8b33988e79494f088/html/dom/elements/global-attributes/the-anchor-attribute-001.tentative.html), with https://github.com/whatwg/html/pull/9144),
 * where the spec is unclear (e.g, [`webdriver/tests/bidi/input/release_actions/sequence_tentative.py`](https://github.com/web-platform-tests/wpt/blob/73fd679bba753839932a62a8b33988e79494f088/webdriver/tests/bidi/input/release_actions/sequence_tentative.py), with https://github.com/w3c/webdriver/issues/1859).

The other thing that stands out is:

```
% rg --glob '*tentative*' 'https?://((bugs\.webkit\.org|bugs\.chromium\.org|crbug\.com|issues\.chromium\.org)/|github\.com/.*/(issue|pull))' --files-without-match | wc -l
    2078
```

Which shows it's very often _not_ obvious "why the test is tentative and what needs to be resolved to make it non-tentative", though that is really a separate issue. (And looking at a few random tests, the commit history often doesn't make it obvious either.)